listen/property-changes.js

```diff
@@ -153,55 +152,49 @@ PropertyChanges.prototype.removeBeforeOwnPropertyChangeListener = function (key,
 };
 
 PropertyChanges.prototype.dispatchOwnPropertyChange = function (key, value, beforeChange) {
-    var descriptor = PropertyChanges.getOwnPropertyChangeDescriptor(this, key);
-
-    if (descriptor.isActive) {
-        return;
-    }
-    descriptor.isActive = true;
-
-    var listeners;
-    if (beforeChange) {
-        listeners = descriptor.willChangeListeners;
-    } else {
-        listeners = descriptor.changeListeners;
-    }
-
-	if (listeners.length === 0) return;
-
-    try {
-        // dispatch to each listener
-        dispatchEach.call(this, listeners, key, value);
-    } finally {
+    var descriptor = PropertyChanges.getOwnPropertyChangeDescriptor(this, key),
+        listeners;
+
+    if (!descriptor.isActive) {
+        descriptor.isActive = true;
+        if (beforeChange) {
+            listeners = descriptor.willChangeListeners;
+        } else {
+            listeners = descriptor.changeListeners;
+        }
+        dispatchEach(listeners, key, value, this);
         descriptor.isActive = false;
```

Comment:
To be clear, an error thrown by a listener invalidates the state of the program thereafter.